spirv_hlsl.cpp

```diff
@@ -3470,8 +3476,8 @@ string CompilerHLSL::to_resource_binding(const SPIRVariable &var)
 		return "";
 
 	uint32_t desc_set =
-	    resource_flags == HLSL_BINDING_AUTO_PUSH_CONSTANT_BIT ? ResourceBindingPushConstantDescriptorSet : 0u;
```

Comment:
You're supposed to set up a binding using the normal API here. `add_hlsl_resource_binding` where you use the ResourceBindingPushConstantDescriptorSet/Binding as a pseudo-binding point. Does that not work? It's fine adding a CLI option for this, but I don't think we need to change the API.

Reply:
Thanks for pointing to `add_hlsl_resource_binding` - I was not aware of this api.
Overall this almost works for me, except stage flags:
```
HLSLResourceBinding push_binding;
push_binding.stage = ???; // can't specify anything until shader is parsed
push_binding.desc_set = ResourceBindingPushConstantDescriptorSet;
push_binding.binding = 0;
push_binding.cbv.register_binding = args.hlsl_push_constant_register_index;
push_binding.cbv.register_space = args.hlsl_push_constant_register_space;
hlsl->add_hlsl_resource_binding(push_binding);
```
Can't specify anything at `push_binding.stage ` until shader is parsed. Fine with my engine, but why `.stage` is needed in the first place? AFAIK spirv-cross processes one shader at a time, so no need to use different settings per stage.

Reply:
You can query the execution mode. The stage there was probably not the best idea ever, but it would be useful if multi-entry point codegen was supported at some point.

Reply:
Thanks!
Already added `HLSLResourceBinding` to my game engine - all works well. 
Since issue is resolved, we can discard PR